lib/tictactoe/board.rb

```diff
@@ -8,29 +9,40 @@ class Board
     DRAW = 'draw'
     ONGOING = 'ongoing'
 
-    def initialize(dim = 3, initial_positions = nil)
+    def initialize(dim = 3, initial_positions = nil, win_case_calculator = nil)
       @dimension = dim
-      @size = @dimension * @dimension
-      initial_positions ||= (EMPTY_MARKER * @size).split('')
+      initial_positions ||= (EMPTY_MARKER * @dimension * @dimension).split('')
       @positions = initial_positions
+      win_case_calculator ||= WinCaseCalculator.new(@dimension)
+      @win_case_calculator = win_case_calculator
```

Comment:
this is a bit weird. should this belong on the board? can we avoid instantiating (and defaulting) this object?